### PR TITLE
Implement new logger

### LINKED ISSUE #1420: Selectively display debug messages in console
Is there a possibility to block some messages for showing up in the mcc console? For example i dont want to see messages that start with test. If yes, how can I implement it. If no, can you open a pull request?
Thanks

### PR DESCRIPTION
**Pull request by ReinforceZwei**
Resolve #1420

### CODE REVIEW

File: MinecraftClient/Resources/config/MinecraftClient.ini [hunk 1]
```diff
@@ -35,11 +35,18 @@ accountlist=accounts.txt           # See README > 'Servers and Accounts file' fo
 serverlist=servers.txt             # See README > 'Servers and Accounts file' for more info about this file
 playerheadicon=true                # Only works on Windows XP-8 or Windows 10 with old console
 exitonfailure=false                # Disable pauses on error, for using MCC in non-interactive scripts
-debugmessages=false                # Please enable this before submitting bug reports. Thanks!
 scriptcache=true                   # Cache compiled scripts for faster load on low-end devices
 timestamps=false                   # Prepend timestamps to chat messages
 autorespawn=false                  # Toggle auto respawn if client player was dead (make sure your spawn point is safe)
 
+# Loggin settings
```
Comment (ORelio) [inline MinecraftClient/Resources/config/MinecraftClient.ini:42]: How about adding a dedicated section like `[Logging]`?

**Review by ORelio — COMMENTED** [review]
Wow, congrats for implementing full logging level as well as implementing sorting for many messages that were already there 💯

I think it does not fully solve #1420 because the issue author @xXjojaXx was seeking to mute other chatbot's debug messages while keeping theirs (or something like that). But anyway this can be addressed in a further pull request, no need to do it right now.

Regarding this PR, here are a few suggestions:

**Comment by ReinforceZwei** [comment]
> mute other chatbot's debug messages

Do I add another message filter for the debug message like chat message?

**Comment by ORelio** [comment]
> Done

Perfect 👌

> Do I add another message filter for the debug message like chat message?

That could work, indeed! With ability to set in whitelist/blacklist mode 👍

**Comment by xXjojaXx** [comment]
Large shoutout @ReinforceZwei for working that fast and reliable.

**Comment by ReinforceZwei** [comment]
The last commit should completed the feature request.

File: MinecraftClient/Settings.cs [hunk 2]
```diff
@@ -285,8 +296,9 @@ public static void LoadSettings(string settingsfile)
                                                 case "enableentityhandling": EntityHandling = str2bool(argValue); break;
                                                 case "inventoryhandling": InventoryHandling = str2bool(argValue); break;
                                                 case "privatemsgscmdname": PrivateMsgsCmdName = argValue.ToLower().Trim(); break;
-                                                case "debugmessages": DebugMessages = str2bool(argValue); break;
                                                 case "autorespawn": AutoRespawn = str2bool(argValue); break;
+                                                // Backward compatible so people can still enable debug with old config format
+                                                case "debugmessages": DebugMessages = str2bool(argValue); break;
```
Comment (ORelio) [inline MinecraftClient/Settings.cs:301]: Good catch! 👍

**Review by ORelio — APPROVED** [review]
Looks good. Congrats! Feel free to merge if you are done 👍

**Pull request merged**